### ISSUE TITLE
Feature: Use DNSLink

### ISSUE THREAD

**Issue by Stebalien** [ProtoSchool/protoschool.github.io#274]
I'm not sure if this is possible but it would be _awesome_ if proto.school could use DNSLink. I'm preparing for some potentially offline presentations and was able to simply "pin" dag.ipfs.io, explore.ipld.io, and cid.ipfs.io, but not proto.school.

**Comment by terichadbourne** [comment]
I don't know a ton about domains, deployment, or DNSLink, but I would guess that this suggestion might be dependent on #18? Does that sound right @mikeal? (We're currently publishing via GitHub Pages, and we need our chapter repos to continue being able to do that.)

**Comment by hacdias** [comment]
@terichadbourne I think this is a must: having ProtoSchool on IPFS itself would be awesome.

If I understand correctly, each chapter can publish to `https://proto.school/<city>`, right? If so, I think it will be a bit harder to put this on IPFS, otherwise it should be rather simple.

**Comment by terichadbourne** [comment]
Yes that's right @hacdias, and we also have the issue where we're using hash-based routing within the whole Vue SPA, which I know can sometimes make it harder to host on IPFS. 

Perhaps another issue would be how localStorage works? If there would be an IPFS-hosted and a "normally"-hosted version of the site, I assume the caching of the user's progress wouldn't be recognized unless they always visited it the same way?

I'm starting to think that this issue might actually be a duplicate of #18 rather than just dependent on it, but I'm not familiar enough with infra stuff to confirm. @hacdias would you mind taking a look? Especially at the discussion starting here: https://github.com/ProtoSchool/protoschool.github.io/issues/18#issuecomment-514362490

If we're really having the same convo in two places, we should consolidate into #18.

**Comment by hacdias** [comment]
@terichadbourne yes, they are kind of related. It seems to me that #18 already encompasses the usage of DNSLink so, yes, I would close one of them in favour of the other.

> hash-based routing within the whole Vue SPA, which I know can sometimes make it harder to host on IPFS.

Not sure if I'm understanding correctly, but Web UI also uses hash-based routing and it works perfectly when hosted on IPFS 😃 

> Perhaps another issue would be how localStorage works? If there would be an IPFS-hosted and a "normally"-hosted version of the site, I assume the caching of the user's progress wouldn't be recognized unless they always visited it the same way?

You are correct!

**Comment by terichadbourne** [comment]
Thanks @hacdias! I'll close this one in favor of continuing conversation in #18.
